From: Mark Baker <mark@mnb.org.uk>
Description: Include old interface to RE::Init() for PCRE 6.x compatibility

Index: pcre-8.30/pcrecpp.cc
===================================================================
--- pcre-8.30.orig/pcrecpp.cc	2012-01-15 19:00:31.000000000 +0100
+++ pcre-8.30/pcrecpp.cc	2012-03-23 11:05:02.219693149 +0100
@@ -80,6 +80,12 @@
 // If the user doesn't ask for any options, we just use this one
 static RE_Options default_options;
 
+// PCRE6.x compatible API
+void RE::Init(const char *c_pat, const RE_Options* options) {
+  const string cxx_pat(c_pat);
+  Init(cxx_pat, options);
+}
+
 void RE::Init(const string& pat, const RE_Options* options) {
   pattern_ = pat;
   if (options == NULL) {
Index: pcre-8.30/pcrecpp.h
===================================================================
--- pcre-8.30.orig/pcrecpp.h	2010-01-02 17:25:49.000000000 +0100
+++ pcre-8.30/pcrecpp.h	2012-03-23 11:05:02.223026534 +0100
@@ -658,6 +658,8 @@
  private:
 
   void Init(const string& pattern, const RE_Options* options);
+  // Old version from PCRE 6.x, for compatibility
+  void Init(const char *pattern, const RE_Options* options);
   void Cleanup();
 
   // Match against "text", filling in "vec" (up to "vecsize" * 2/3) with
Index: pcre-8.30/pcretest.c
===================================================================
--- pcre-8.30.orig/pcretest.c	2012-02-04 15:35:53.000000000 +0100
+++ pcre-8.30/pcretest.c	2012-03-23 11:05:02.233026691 +0100
@@ -2152,7 +2152,7 @@
 {
 FILE *infile = stdin;
 const char *version;
-int options = 0;
+long int options = 0;
 int study_options = 0;
 int default_find_match_limit = FALSE;
 int op = 1;
